security/src/main/java/io/micronaut/security/authentication/provider/AuthenticationProviderUtils.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.authentication.provider;
+
+import io.micronaut.context.BeanContext;
+import io.micronaut.context.BeanRegistration;
+import io.micronaut.core.annotation.Blocking;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.inject.ExecutableMethod;
+import io.micronaut.security.authentication.AuthenticationRequest;
+
+import java.lang.annotation.Annotation;
+
+/**
+ * Utility class to check whether {@link AuthenticationProvider#authenticate(Object, AuthenticationRequest)} is annotated with {@link Blocking}.
+ */
+@Internal
+public final class AuthenticationProviderUtils {
```

Comment:
does this need to be `public`?

Reply:
I moved it to the same package as `Authenticator` to make it package private. 

https://github.com/micronaut-projects/micronaut-security/pull/1526/commits/69c09f7bae7680b0e2b9ec0d64f7c72d79a0790a